File: best_present_1080.go
// https://codeforces.com/problemset/problem/1080/B
package main

import (
	"fmt"
	"strconv"
	"bufio"
	"strings"
	"os"
)

func formula(pos int64) int64 {
	if pos % 2 == 0 {
		return int64(pos/2)
	}
	return int64(pos-1)/2 - pos
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	numInputs, _ := strconv.ParseInt(scanner.Text(), 10, 32)
	for numInputs > 0 {
		scanner.Scan()
		inputs := strings.Split(scanner.Text(), " ")
		start, _ := strconv.ParseInt(inputs[0], 10, 32)
		last, _ := strconv.ParseInt(inputs[1], 10, 32)
		fmt.Println(int64(formula(last)) - int64(formula(start-1)))
		numInputs--
	}
}

File: best_present_1080.py

import sys

def calculate(a, b):
    return a*(a+1) - (b*b)

def formula(pos):
    ceil, floor = int(pos/2) + 1, int(pos/2)
    if pos % 2 == 0:
        return calculate(ceil, floor)
    else:
        return calculate(floor, ceil)

if __name__ == "__main__":
    numLines = int(input())
    for i in range(numLines):
        inputs = input().split(" ")
        e, s = int(inputs[0]), int(inputs[1])
        print(int(formula(e) - formula(s-1)))